File: java/src/test/java/com/auth/HawkAuthTest.java
package com.auth;

import com.auth.base.Authentication;
import com.auth.hawk.HawkAuth;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpMethod;
import org.springframework.http.HttpStatus;
import org.springframework.http.RequestEntity;

import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

class HawkAuthTest extends TestUtils {
    private Authentication auth;

    @BeforeEach
    void setUp() {
        super.setUp();
        auth = new HawkAuth("test-hawk-id", "test-hawk-key");
    }

    @Test
    void testValidHawkCredentials() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Hawk id=\"test-hawk-id\"," +
            "ts=\"1234567890\"," +
            "nonce=\"test-nonce\"," +
            "mac=\"test-mac\"");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.OK, false);
    }

    @Test
    void testInvalidHawkId() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Hawk id=\"wrong-id\"," +
            "ts=\"1234567890\"," +
            "nonce=\"test-nonce\"," +
            "mac=\"test-mac\"");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testMissingAuthorizationHeader() {
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, null);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testMalformedAuthorizationHeader() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Hawk");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testInvalidScheme() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Basic dGVzdC11c2VyOnRlc3QtcGFzc3dvcmQ=");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testGetType() {
        assertAuthType(auth, "hawk");
    }
} 

File: php/tests/HawkAuthTest.php
<?php

namespace Tests\Auth;

use Auth\HawkAuth;

class HawkAuthTest extends TestUtils
{
    private $auth;

    protected function setUp(): void
    {
        parent::setUp();
        $this->auth = new HawkAuth('test-hawk-id', 'test-hawk-key', 'sha256');
    }

    public function testValidHawkCredentials(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Hawk id="test-hawk-id",' .
                'ts="1234567890",' .
                'nonce="test-nonce",' .
                'mac="test-mac",' .
                'hash="test-hash"'
        ]);
        $this->runAuthTest($this->auth, $request, 200, false);
    }

    public function testInvalidHawkID(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Hawk id="wrong-hawk-id",' .
                'ts="1234567890",' .
                'nonce="test-nonce",' .
                'mac="test-mac",' .
                'hash="test-hash"'
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testMissingAuthorizationHeader(): void
    {
        $request = $this->createTestRequest('GET');
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testMalformedAuthorizationHeader(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Hawk'
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testInvalidScheme(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Basic dGVzdC11c2VyOnRlc3QtcGFzc3dvcmQ='
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testGetType(): void
    {
        $this->assertAuthType($this->auth, 'hawk');
    }
} 